elements/src/components/pos-dialog/test/pos-dialog.spec.tsx

```diff
@@ -0,0 +1,82 @@
+import { newSpecPage } from '@stencil/core/testing';
+import { screen } from '@testing-library/dom';
+import { PosDialog } from '../pos-dialog';
+
+describe('pos-dialog', () => {
+  it('renders a dialog with the given slot content', async () => {
+    const page = await newSpecPage({
+      components: [PosDialog],
+      html: `<pos-dialog><span slot="dialog-title">Title</span><span slot="dialog-content">Content</span></pos-dialog>`,
+    });
+    expect(page.root).toEqualHtml(`
+<pos-dialog>
+    <mock:shadow-root>
+        <dialog>
+            <header>
+                <span id="title">
```

Comment:
I was wondering about the double <span> we are introducing here (the one with id "title" and the one that is "slotted in". I guess the title element should actually be a `h2`or something. What do you think?

Reply:
Or we could use a `hgroup`here, then the user of  pos-dialog can decide wich h-level to use and whether to also add a subheading or something like that.

Reply:
The reason for the double is entirely historical.

The original css uses the title id
https://github.com/pod-os/PodOS/blob/2aa9c2ce94fe742fb2e237409978b3881ebf2d7d/elements/src/components/pos-add-new-thing/pos-add-new-thing.css#L34

The slot is therefore nested within the span with that id.

We can therefore remove the double by changing the css to:
```css
dialog slot[name='title']::slotted(*) {
```

We would then have (ignoring the shadow dom):
```html
<pos-dialog>
        <dialog>
            <header>
                <span slot="title">
                    Title
                </span>
                <button tabindex="-1" id="close" title="Close">
                    <ion-icon name="close-outline"></ion-icon>
                </button>
            </header>
            <span slot="content">Content</span>
        </dialog>
</pos-dialog>
```

Switching to `h2` or `hgroup` could make greater use of `#title` but it would be a change in styling from the current.

I've made the change that removes the double but keeps current style, but also happy for you to make an executive decision. I'm very bad at style decisions :-)

Reply:
we can leave it for now, it has been a span before